src/__tests__/index.spec.js

```diff
@@ -1,64 +1,42 @@
 import CharacterProfileBuilder from '../examples/CharacterProfileBuilder';
-import TelevisionShowBuilder from '../examples/TelevisionShowBuilder';
 import mockProvider from '../templates/__tests__/mockProvider';
-import * as tcombBuilder from '../index';
-
-const CHARACTER_PROFILE = 'CHARACTER_PROFILE';
-const TELEVISION_SHOW = 'TELEVISION_SHOW';
+import {
+  BaseBuilder,
+  primitives,
+  validators,
+  widgets,
+} from '../index';
 
 describe('index sanity test', () => {
-  const input = [
-    {
-      builder: CharacterProfileBuilder,
-      name: CHARACTER_PROFILE,
-    },
-    {
-      builder: TelevisionShowBuilder,
-      name: TELEVISION_SHOW,
-    },
-  ];
-
-  describe('create', () => {
-    it('generates the correct API structure', () => {
-      const survey = tcombBuilder.create(input, mockProvider);
-      expect(survey[CHARACTER_PROFILE].options).to.not.be.undefined;
-      expect(survey[CHARACTER_PROFILE].type).to.not.be.undefined;
-      expect(survey[TELEVISION_SHOW].options).to.not.be.undefined;
-      expect(survey[TELEVISION_SHOW].type).to.not.be.undefined;
-    });
-  });
-
-  describe('createTypes', () => {
-    it('generates the correct API structure', () => {
-      const types = tcombBuilder.createTypes(input);
-      expect(types[CHARACTER_PROFILE]).to.not.be.undefined;
-      expect(types[TELEVISION_SHOW]).to.not.be.undefined;
-    });
+  describe('provider can be accessed from a builder', () => {
+    const builder = CharacterProfileBuilder.setLazyTemplateProvider(mockProvider);
+    expect(() => builder.getType()).to.not.throw();
+    expect(() => builder.getOptions()).to.not.throw();
   });
 
   describe('primitives', () => {
     it('are exported', () => {
-      expect(tcombBuilder.primitives.CheckboxBuilder).to.not.be.undefined;
-      expect(() => tcombBuilder.primitives.CheckboxBuilder.setLabel('foobar')).to.not.throw();
+      expect(primitives.CheckboxBuilder).to.not.be.undefined;
```

Comment:
Does undefined not have to be a function? I thought Enzyme switched everything to be a function call?

Reply:
It's not unless `dirty-chai` is used, which is planned but out of scope for this PR.

https://github.com/cadre/tcomb-builder/issues/1